### ISSUE TITLE
Passing incorrect Provider in subjects gives build successful message

### ISSUE THREAD

**Issue by arya-hemanshu** [FutureCitiesCatapult/TomboloDigitalConnector#391]
### Description
When an incorrect provider is passed, user gets build successful message, instead of quitting with error/exception "Invalid Provider"
 ```
 "subjects": [
      {
        "subjectType": "localAuthority",
        "provider": "doesnotExists"
      }
    ],
```

### Error log
```
C02Q99ZZG8WN:TomboloDigitalConnector hemanshu$ gradle runExport   -PdataExportSpecFile='/Users/hemanshu/Desktop/json_dump_test.json'   -PoutputFile='/Users/hemanshu/Desktop/thisisatest.json' -PforceImports='uk.org.tombolo.importer.ons.CensusImporter'

> Task :runExport
log4j:WARN No appenders could be found for logger (org.jboss.logging).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
2017-12-12 12:32:14.141 [main] WARN  n.s.e.config.ConfigurationFactory - No configuration found. Configuring ehcache from ehcache-failsafe.xml  found in the classpath: jar:file:/Users/hemanshu/.gradle/caches/modules-2/files-2.1/net.sf.ehcache/ehcache/2.10.3/cf74f9a4a049f181833b147a1d9aa62159c9d01d/ehcache-2.10.3.jar!/ehcache-failsafe.xml
2017-12-12 12:32:14.574 [main] INFO  uk.org.tombolo.AbstractRunner - Setting file download cache: /tmp
2017-12-12 12:32:14.661 [main] INFO  org.ehcache.core.EhcacheManager - Cache 'fields' created in EhcacheManager.
2017-12-12 12:32:15.132 [main] INFO  u.o.t.importer.AbstractImporter - Skipped importing uk.org.tombolo.importer.ons.OaImporter:localAuthority as this import has been completed previously
2017-12-12 12:32:15.139 [main] INFO  u.org.tombolo.importer.DownloadUtils - Fetching local file: e18d58ee-6c0e-37d0-81df-2b877dd5f99c.json
2017-12-12 12:32:15.285 [main] INFO  u.o.t.importer.AbstractImporter - Importing uk.org.tombolo.importer.ons.CensusImporter:qs106ew
2017-12-12 12:32:15.287 [main] INFO  u.org.tombolo.importer.DownloadUtils - Fetching local file: a7104fb5-322e-388e-aa54-160a9717c2bd.csv
2017-12-12 12:32:15.357 [main] INFO  u.org.tombolo.importer.DownloadUtils - Fetching local file: 7abc157d-8ea2-3808-a9d3-e8a7efec4395.csv
2017-12-12 12:32:15.915 [main] INFO  u.o.t.importer.AbstractImporter - Preparing to write a batch of 735 timed values ...
2017-12-12 12:32:16.290 [main] INFO  u.o.t.importer.AbstractImporter - Total timed values written: 735
2017-12-12 12:32:16.310 [main] INFO  u.o.t.importer.AbstractImporter - Imported 0 subjects, 0 fixed values and 735 timedValues
2017-12-12 12:32:16.312 [main] INFO  u.o.t.execution.DataExportEngine - Exporting ...
2017-12-12 12:32:16.318 [main] INFO  u.o.tombolo.exporter.GeoJsonExporter - Exporting 0 subjects
2017-12-12 12:32:16.318 [main] INFO  u.o.tombolo.exporter.GeoJsonExporter - Exported 0 subjects


BUILD SUCCESSFUL in 4s
3 actionable tasks: 1 executed, 2 up-to-date
```

**Comment by lorenaqendro** [comment]
See also #319.

**Comment by lorenaqendro** [comment]
Can be closed as merged in #433.
